internal/k8s/services_api.go

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2020-2021 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package k8s
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	servicesapi "sigs.k8s.io/service-apis/apis/v1alpha1"
+
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/nodes"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/status"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+// Services API related functions. Parking the functions on this file instead of creating a new one since most of the functionality is same with v1alpha1pre1
+
+func InformerStatusUpdatesForSvcApiGateway(key string, gateway *servicesapi.Gateway) {
+	gwStatus := gateway.Status.DeepCopy()
+	defer status.UpdateSvcApiGatewayStatusObject(gateway, gwStatus)
+	status.InitializeSvcApiGatewayConditions(gwStatus, &gateway.Spec, false)
+	gwClassObj, err := lib.GetSvcAPIInformers().GatewayClassInformer.Lister().Get(gateway.Spec.GatewayClassName)
+	if err != nil {
+		status.UpdateSvcApiGatewayStatusGWCondition(gwStatus, &status.UpdateSvcApiGWStatusConditionOptions{
+			Type:    "Pending",
```

Comment:
gateway status conditions changed from `Pending` to `Scheduled` (GatewayConditionScheduled)
This essentially also completes migration of gateway status conditions (leaving listener status conditions for now)

Reply:
Thx. Done.